lib/npc-generation/professions.ts

```diff
@@ -91,7 +92,11 @@ export const professions: Record<string, Profession> = {
           relationship: 'lord',
           reciprocalRelationship: 'labourer',
           exclusions (town, npc) {
-            return Object.values(npc.relationships).includes('lord')
+            if (town.npcRelations) {
+              if (town.npcRelations[npc.key]) {
+                return town.npcRelations[npc.key].map((r: NpcRelationship): string => { return r.relation }).includes('lord')
+              }
+            }
```

Comment:
Perhaps it would be easier to construct a function which does this? . A function which checks for a relationship with the matching string would be quite useful, and a call to `checkForRelationship` would be a little more readable.